src/coordination/coordinator_instance.cpp

```diff
@@ -1056,63 +679,38 @@ auto CoordinatorInstance::UnregisterReplicationInstance(std::string_view instanc
     return UnregisterInstanceCoordinatorStatus::LOCK_OPENED;
   }
 
-  auto const name_matches = [&instance_name](ReplicationInstanceConnector const &instance) {
-    return instance.InstanceName() == instance_name;
-  };
+  auto const name_matches = [&instance_name](auto &&instance) { return instance.InstanceName() == instance_name; };
 
   auto inst_to_remove = std::ranges::find_if(repl_instances_, name_matches);
   if (inst_to_remove == repl_instances_.end()) {
     return UnregisterInstanceCoordinatorStatus::NO_INSTANCE_WITH_NAME;
   }
 
-  auto const is_current_main = [this](ReplicationInstanceConnector const &instance) {
+  auto const is_current_main = [this](auto &&instance) {
     return raft_state_->IsCurrentMain(instance.InstanceName()) && instance.IsAlive();
   };
 
   if (is_current_main(*inst_to_remove)) {
     return UnregisterInstanceCoordinatorStatus::IS_MAIN;
   }
 
-  if (!raft_state_->AppendOpenLock()) {
-    return UnregisterInstanceCoordinatorStatus::FAILED_TO_OPEN_LOCK;
+  if (!raft_state_->AppendUnregisterReplicationInstanceLog(instance_name)) {
```

Comment:
How can we append without opening/closing the lock?

Reply:
because we only do one action. Locks are important when having multiple appends

Reply:
Fine while logic is protected by the mtx